contrib/holland-commvault/holland_commvault/commvault.py

```diff
@@ -94,7 +105,35 @@ def main():
     args.dry_run = 0
     args.no_lock = 0
     largs = args.bksets
-    if run(args, largs):
+    spool = HOLLANDCFG.lookup('holland.backup-directory')
+    status_file = '%s/%s/newest/job_%s' % (spool, args.bksets[0], args.job)
+    pid_name = 'holland_commvault_%s' % args.job
+    pid_location = '/var/run/%s.pid' % pid_name
+    try:
+        with PidFile(pid_name):
+            ret = 0
+            if run(args, largs):
+                ret = 1
+            status = open(status_file, 'w')
+            status.write(str(ret))
+            status.close()
+            return ret
+    except PidFileAlreadyLockedError:
+        pid_file = open(pid_location, 'r')
+        pid = pid_file.read()
+        pid_file.close()
+
+        logging.info("Holland (commvault agent) is already running, waiting for the pid %s", pid)
+        while os.path.isfile(pid_location):
```

Comment:
Do we want to set a reasonable timeout for this? I'm not sure if there's a scenario where commvault would run for so long that we'd want Holland to error out waiting for it and maybe cause a failure so someone looks into it.

Reply:
I've added a 10 hour timeout